src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTokenExtensions.cs

```diff
@@ -99,91 +99,74 @@ public static bool IsBeginningOfStatementContext(this SyntaxToken token)
             // lock (expr)
             //   |
 
-            // for ( ; ; Foo(), |
```

Comment:
Yes.  it does not apply.  That's not a statement location (i.e. you can't type 'return' there).  This is also why there's no case below that handles hte .Comma case.